File: abc285/b.cpp
#include<iostream>
#include<string>
using namespace std;

int main(){
    int N;
    string S;
    int l;
    
    cin >> N;
    cin >> S;
    int prev = N-1;
    for (int i = 1; i<N; i++){
        l = 0;
        bool invalid = false;
        while(l<= prev){
            for(int k = 1; k<l+1; k++){
                if (S[k-1] == S[k+i-1]){
                    invalid = true;
                    break;
                }
            }
            if (invalid) break;
            l++;
        }
        prev = l-1;
        cout << l-1 << endl;
    }
    
    return 0;
}

File: abc285/b.py

N = int(input())
S = input()

for i in range(1, N):
    
    for k in range(1, N-i+1):
        if S[k-1] != S[k+i-1]:
            pass
        else:
            k -= 1
            break
        #l += 1
    #if invalid:
    #    break
    #l += 1
    
    print(k)